File: Rest.php
<?php 
     //tipo de archivo que se va a enviar
     $tipo = "xml";
     //rut de la empresa
     $rut = "11111111-1";
     //url del web service 
     $url='https://develop.ingefactura.cl/wsRest/web/com/dte';
     //token de acceso (Se debe pedir a soporte de ingefactura)
     $token='xxxxxxxxxxxxxxxxxxxxxx';
     // se toma la ruta del archivo
     $file_name = realpath('ruta del archivo');
     $fichero = file_get_contents($file_name);     

     $ch = curl_init();

     curl_setopt($ch, CURLOPT_URL, $url);
     curl_setopt($ch, CURLOPT_POST, true);
     curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
     curl_setopt($ch, CURLOPT_POSTFIELDS, $fichero);
     curl_setopt($ch, CURLOPT_HTTPHEADER, array('Authorization:  Bearer ' . $token ,
                                                     'Content-Type: application/xml' ,
                                                            'rut: ' .$rut   ,
                                                                 'tipo:'.$tipo));       
     //se ejecuta y trae respuesta 
     $response = curl_exec($ch);
     //se cierra 
     curl_close($ch);

     //se toma la respuesta 
     $data = htmlentities($response);
     //se imprime 
     echo $data;

     ?>

File: Rest.java
package test;

import java.io.DataOutputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.InputStream;
import java.io.StringWriter;
import java.net.URL;
import java.security.SecureRandom;
import java.security.Security;
import java.security.cert.CertificateException;
import java.security.cert.X509Certificate;
import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.HttpsURLConnection;
import javax.net.ssl.SSLContext;
import javax.net.ssl.SSLSession;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;
import org.apache.commons.io.IOUtils;

public class TestPost {

    public static void main(String[] args) {
        
       try{
                //se selecciona el archivo
                File file = new File("UrlArchivo");
                // 
                InputStream input = new FileInputStream(file);
                //documento convertido a string
                String archivo = FileUtils.readFileToString(file);
                //url del web service
                HttpsURLConnection connection = (HttpsURLConnection) new URL("https://develop.ingefactura.cl/wsRest/web/com/dte").openConnection(); 
                                   
                //parametros de envio
                connection.setDoOutput(true);
                connection.setDoInput(true);
                //metodo de envio
                connection.setRequestMethod("POST");
                connection.addRequestProperty("Content-Type", "application/xml");
                connection.addRequestProperty("rut", "Rut Empresa");
                //envio del token
                connection.addRequestProperty("Authorization","Bearer "+"Token");
                //tipo de archivo a enviar
                connection.addRequestProperty("tipo", "Tipo de archivo");
                DataOutputStream os = new DataOutputStream(connection.getOutputStream());
                os.writeBytes(archivo);
                os.close();
                 
                //respuesta xml del web service 
                InputStream salida = connection.getInputStream();
                String xmlRespuesta = IOUtils.toString(salida,"UTF-8");
                System.out.println(xmlRespuesta);
       
       } catch (FileNotFoundException ex) {
            ex.printStackTrace();
       } catch (Exception ex) {
            ex.printStackTrace();
        }
       
    }
    